### PR TITLE
feature: adds sentry to the app

### PR DESCRIPTION
**Pull request by geclos**

### CODE REVIEW

File: apps/web/next.config.mjs [hunk 1]
```diff
@@ -29,4 +31,49 @@ const nextConfig = {
...
+    hideSourceMaps: true,
+
+    // Automatically tree-shake Sentry logger statements to reduce bundle size
+    disableLogger: true,
+
+    // Enables automatic instrumentation of Vercel Cron Monitors. (Does not yet work with App Router route handlers.)
+    // See the following for more information:
+    // https://docs.sentry.io/product/crons/
+    // https://vercel.com/docs/cron-jobs
+    automaticVercelMonitors: true,
```
Comment (andresgutgon) [inline apps/web/next.config.mjs:73]: False this no?

File: apps/web/next.config.mjs [hunk 2]
```diff
@@ -29,4 +31,49 @@ const nextConfig = {
...
+    // Automatically annotate React components to show their full name in breadcrumbs and session replay
+    reactComponentAnnotation: {
+      enabled: true,
+    },
+
+    // Route browser requests to Sentry through a Next.js rewrite to circumvent ad-blockers.
+    // This can increase your server load as well as your hosting bill.
+    // Note: Check that the configured route will not match with your Next.js middleware, otherwise reporting of client-
+    // side errors will fail.
+    tunnelRoute: '/monitoring',
```
Comment (andresgutgon) [inline apps/web/next.config.mjs:61]: Review this according to comment

File: apps/web/next.config.mjs [hunk 3]
```diff
@@ -29,4 +31,49 @@ const nextConfig = {
...
+if (process.env.SENTRY_ORG && process.env.SENTRY_PROJECT) {
+  config = withSentryConfig(nextConfig, {
+    // For all available options, see:
+    // https://github.com/getsentry/sentry-webpack-plugin#options
+
+    org: process.env.SENTRY_ORG,
+    project: process.env.SENTRY_PROJECT,
+
+    // Only print logs for uploading source maps in CI
+    silent: !process.env.CI,
```
Comment (andresgutgon) [inline apps/web/next.config.mjs:44]: We don't have this env variable unless is setup automatically by GH actions?

File: apps/web/src/app/global-error.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+'use client'
```
Comment (andresgutgon) [inline apps/web/src/app/global-error.tsx:1]: We already have a global error I think

File: apps/web/next.config.mjs (hunk 1 above)
Comment (geclos) [reply apps/web/next.config.mjs:73]: this is noop if we don't deploy to vercel (automatically generated by sentry)

File: apps/web/next.config.mjs (hunk 3 above)
Comment (geclos) [reply apps/web/next.config.mjs:44]: it does

File: apps/web/src/app/global-error.tsx (hunk 4 above)
Comment (geclos) [reply apps/web/src/app/global-error.tsx:1]: Nein we have an error for private pages

Comment (andresgutgon) [reply apps/web/src/app/global-error.tsx:1]: And Sentry catch errors that are catched by Private pages ErrorBoundary?

Comment (geclos) [reply apps/web/src/app/global-error.tsx:1]: yes this is just to make sure sentry also catches any unexpected error outside that boundary

**Review by andresgutgon — APPROVED** [review]

**Pull request merged**